packages/flutter_test/lib/src/binding.dart

```diff
@@ -85,6 +84,30 @@ const Size _kDefaultTestViewportSize = Size(800.0, 600.0);
 /// It tracks status of data sent across the Flutter platform barrier, which is
 /// useful for testing frameworks to monitor and synchronize against the
 /// platform messages.
+///
+/// ## Messages from the framework to the platform
+///
+/// Messages are sent from the framework to the platform via the
+/// [send] method.
+///
+/// To intercept a message sent from the framework to the platform,
+/// consider using [setMockMessageHandler],
+/// [setMockDecodedMessageHandler], and [setMockMethodCallHandler]
+/// (see also [checkMockMessageHandler]).
+///
+/// To wait for all pending framework-to-platform messages, the
+/// [platformMessagesFinished] getter provides an appropriate
+/// [Future]. The [pendingMessageCount] getter returns the current
+/// number of outstanding messages.
+///
+/// ## Messages from the platform to the framework
+///
+/// The platform sends messages via the [ChannelBuffers] API. Mock
+/// messages can be sent to the framework using
+/// [handlePlatformMessage].
+///
+/// Listeners for these messages are configured using [setMessageHandler].
+// TODO(ianh): move this to its own file
```

Comment:
it would have made the diff harder to review. that was before i realised i was going to have to have multiple commits anyway, though, so i could always add it here in its own commmit.